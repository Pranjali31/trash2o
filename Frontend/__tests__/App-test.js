import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import CustomButton from '../src/components/atoms/customButton/index';

props = {onPress : () => {}, children : ""}
test("index Component Loading", () => {
    render(<CustomButton props={props} />)

})
test("index component snapshot", () => {
    let component = render(<CustomButton props={props} />).toJSON();
    expect(component).toMatchSnapshot();
})


describe("index Testing component", () => {
    var component;
    beforeEach(() => {
        component = render(<CustomButton props={props} />);

    })

    test("Testing Styles", () => {
        const index = component.toJSON();
        // console.log(JSON.stringify(homescreen));
        let Styles = index.children[0].children[0].props.style;
        expect(Styles).toStrictEqual({
            color: 'white',
            fontSize: 18,
        });
    });

        test("Testing Styles", () => {
            const index = component.toJSON();
            // console.log(JSON.stringify(homescreen));
            let Styles = index.children[0].props.style;
            expect(Styles).toStrictEqual({
                borderRadius: 25,
                paddingHorizontal: 30,
                paddingVertical: 12,
            });
    })

})