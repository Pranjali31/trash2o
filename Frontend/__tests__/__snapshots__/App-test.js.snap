// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index component snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 25,
        "paddingHorizontal": 30,
        "paddingVertical": 12,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 18,
        }
      }
    />
  </View>
</View>
`;
